ext/groovy/src/main/java/org/mvcspec/ozark/ext/groovy/GroovyViewEngine.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2017 Ivar Grimstad (ivar.grimstad@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mvcspec.ozark.ext.groovy;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.mvc.engine.ViewEngineContext;
+import javax.mvc.engine.ViewEngineException;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+
+import org.codehaus.groovy.control.CompilationFailedException;
+import org.mvcspec.ozark.engine.ViewEngineBase;
+import org.mvcspec.ozark.engine.ViewEngineConfig;
+
+import groovy.lang.Writable;
+import groovy.text.Template;
+import groovy.text.markup.MarkupTemplateEngine;
+
+/**
+ * @author Daniel Dias
+ */
+@ApplicationScoped
+public class GroovyViewEngine extends ViewEngineBase {
+
+    @Inject
+    @ViewEngineConfig
+    private MarkupTemplateEngine markupTemplateEngine;
+
+    @Inject
+    private ServletContext servletContext;
+
+    public boolean supports(String view) {
+        return view.endsWith(".tpl");
+    }
+
+    public void processView(ViewEngineContext context) throws ViewEngineException {
+
+        Map<String, Object> model = new HashMap<>(context.getModels());
+        model.put("request", context.getRequest(HttpServletRequest.class));
+        Charset charset = resolveCharsetAndSetContentType(context);
+        try (Writer writer = new OutputStreamWriter(context.getOutputStream(), charset);
+            InputStream resourceAsStream = servletContext.getResourceAsStream(resolveView(context));
+            InputStreamReader in = new InputStreamReader(resourceAsStream, "UTF-8");) {
+            Template template = markupTemplateEngine.createTemplate(in);
+            Writable output = template.make(model);
+            output.writeTo(writer);
+        } catch (IOException | CompilationFailedException | ClassNotFoundException e) {
```

Comment:
I'm just wondering: Which method is throwing `ClassNotFoundException` here? Or can we remove this exception type here?

Reply:
I see this . 

this line :  markupTemplateEngine.createTemplate(in); and what throws the exception .

In this case, I can create a separate catch for this exception.

Reply:
No, that's fine. The exception looks weird, but it the `MarkupTemplateEngine` throws that exception it is fine to catch it together with the other ones.

Reply:
Do you think that would be good?   

` } catch (IOException | CompilationFailedException e) {
            throw new ViewEngineException(e);
        }  catch (ClassNotFoundException e) {
			e.printStackTrace();
		}`

Reply:
The original approach where you catch them all and re-throw as VewEngineException is better. This one would just swallow it and you wouldn't know what happened...

Reply:
I'll do the update for The original approach .